effekt/jvm/src/main/scala/effekt/Runner.scala

```diff
@@ -323,18 +327,52 @@ object LLVMRunner extends Runner[String] {
     val llc = llcCmd.getOrElse(missing("llc"))
     val opt = optCmd.getOrElse(missing("opt"))
 
-    exec(opt, llPath, "-S", "-O2", "-o", optPath)
-    exec(llc, "--relocation-model=pic", optPath, "-filetype=obj", "-o", objPath)
-
     val gccMainFile = (C.config.libPath / ".." / "llvm" / "main.c").unixPath
     val executableFile = basePath
-    var gccArgs = Seq(gcc, gccMainFile, "-o", executableFile, objPath) ++ linkedLibraries
 
-    if (C.config.debug()) gccArgs ++= Seq("-g", "-Wall", "-Wextra", "-Werror")
-    if (C.config.valgrind()) gccArgs ++= Seq("-O0", "-g")
-    else if (C.config.debug()) gccArgs ++= Seq("-fsanitize=address,undefined", "-fstack-protector-all")
-
-    exec(gccArgs: _*)
+    if (useLTO) {
+      // Convert to bitcode with aggressive optimizations
+      exec(opt, llPath, "-O3", "-o", bcPath)
+
+      var gccArgs = Seq(gcc, gccMainFile, bcPath, "-o", executableFile) ++ linkedLibraries
```

Comment:
This likely only works on Clang, but it seems simpler than optimising, and then compiling to an object file...
Since we mandate LLVM anyway, I'd be willing to drop GCC support... ;)